src/api/OpenWith.js

```diff
@@ -20,6 +20,101 @@ class OpenWith extends Base {
 
         return `${this.getBaseApiUrl()}/files/${id}/open_with_integrations`;
     }
+
+    /**
+     * Gets Open With integration data
+     *
+     * @param {string} fileId - Box file ID
+     * @param {Function} successCallback - Success callback
+     * @param {Function} errorCallback - Error callback
+     * @return {Array<OpenWithIntegrationItem>} formatted open with integrations
+     */
+    getOpenWithIntegrations(fileId: string, successCallback: Function, errorCallback: Function) {
+        this.get({
+            id: fileId,
+            successCallback: (openWithIntegrations) => {
+                const formattedOpenWithIntegrations = this.formatOpenWithData(openWithIntegrations);
+                this.fetchAppIntegrations(formattedOpenWithIntegrations, successCallback, errorCallback);
+            },
+            errorCallback
+        });
+    }
+
+    /**
+     * Formats data conviniently for the client
+     *
+     * @param {OpenWithOntegrations} openWithOntegrations - The available Open With integrations
+     * @return {Array<OpenWithIntegrationItem>} formatted open with integrations
+     */
+    formatOpenWithData(openWithIntegrations: OpenWithIntegrations): Array<any> {
+        // Eventually, a default integration will be returned outside of the items array,
+        // which should be inserted into this list of integrations.
+
+        const { items } = openWithIntegrations;
+        // Fix the API response by converting the app integration ID to a string
+        const integrations = items.map((item) => ({
+            ...item,
+            app_integration: {
+                id: item.app_integration.id.toString(),
+                type: 'app_integration'
+            }
+        }));
+
+        // Sort integrations by display_order
+        return integrations.sort(
+            (integrationA: OpenWithIntegrationItem, integrationB: OpenWithIntegrationItem) =>
+                integrationA.display_order - integrationB.display_order
+        );
+    }
+
+    /**
+     * Fetch app intgrations info needed to render.
+     *
+     * @param {Array<OpenWithIntegrationItem>} openWithOntegrations - The available Open With integrations
+     * @return {void}
+     */
+    fetchAppIntegrations = async (
+        openWithIntegrations: Array<OpenWithIntegrationItem>,
+        successCallback: Function,
+        errorCallback: Function
+    ) => {
+        const appIntegrationsAPI = new AppIntegrations(this.options);
```

Comment:
Is there a reason why you aren't using ```APIFactory?```

Reply:
I followed your convention from Feed.js (one endpoint calling another), but I might as well just create my own factory inside of this endpoint.

Reply:
As discussed, disregard the above

Reply:
We followed up offline about this. The idea is to not use the factory but set this as a property so that it will be destroyed later.